osa/nightsummary/tests/test_extract.py

```diff
@@ -28,3 +28,27 @@ def test_build_sequences(sequence_list):
     extracted_seq_list = build_sequences(date=obs_date)
     for sequence, extracted_seq in zip(sequence_list, extracted_seq_list):
         assert sequence.run == extracted_seq.run
+
+
+def test_no_calib_found(run_summary_file_no_calib, run_summary_file_no_calib2):
+    """Test that for a day with no PEDCALIB run, the PEDCALIB
+     run of another previous day is taken."""
+    from osa.nightsummary.extract import extract_runs
+    from osa.nightsummary.nightsummary import run_summary_table
+    from osa.configs import options
+
+    assert run_summary_file_no_calib.exists()
+    assert run_summary_file_no_calib2.exists()
+
+    date = datetime.fromisoformat("2022-09-23")
+    options.date = date
```

Comment:
There is a problem with mixing dates in `options.date` as it will be shared among all tests in the current approach. You could define `options.date = datetime.fromisoformat("2022-09-23")` and then right before ending the function, you define again `datetime.fromisoformat("2020-01-17")` so the other tests keep assuming that initial date.

In the long-term, it would be better to not depend on these globally defined variables and define them locally every time they are used.